Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41840&attempt_id=d8dc9700-a9be-4b6a-9f9c-b97691ba9dc5)

Attempted: gshogi

<details><summary>Partial log (click to expand)</summary><p>

```
engine/gnushogi.h:164:13: error: conflicting types for 'setlinebuf'
extern void setlinebuf(FILE *__stream);
            ^
/nix/store/xn55wd7nimdi085kji54mrrqqnqvcasz-Libsystem-osx-10.11.6/include/stdio.h:461:6: note: previous declaration is here
int      setlinebuf(FILE *);
         ^
1 error generated.
error: command 'clang' failed with exit status 1
builder for '/nix/store/pbnqxg9zgcy07947jxhqwkg40iv2lz50-gshogi-0.5.1.drv' failed with exit code 1
error: build of '/nix/store/pbnqxg9zgcy07947jxhqwkg40iv2lz50-gshogi-0.5.1.drv' failed
```
</p></details>

--- a/pkgs/games/gshogi/default.nix
+++ b/pkgs/games/gshogi/default.nix
@@ -1,0 +1,37 @@
+{ stdenv, buildPythonApplication, fetchFromGitHub
+, gtk3, gobjectIntrospection
+, wrapGAppsHook, python3Packages }:
+
+buildPythonApplication rec {
+  pname = "gshogi";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "johncheetham";
+    repo = "gshogi";
+    rev = "v${version}";
+    sha256 = "06vgndfgwyfi50wg3cw92zspc9z0k7xn2pp6qsjih0l5yih8iwqh";
+  };
+
+  doCheck = false;  # no tests available
+
+  buildInputs = [
+    gtk3
+    gobjectIntrospection
+  ];
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  propagatedBuildInputs = with python3Packages; [
+    pygobject3
+    pycairo
+  ];
+
+  meta = with stdenv.lib; {
+    description = "A graphical implementation of the Shogi board game, also known as Japanese Chess";
+    homepage = http://johncheetham.com/projects/gshogi/;
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.ciil ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -19308,6 +19308,8 @@ with pkgs;
 
   gogui = callPackage ../games/gogui {};
 
+  gshogi = python3Packages.callPackage ../games/gshogi {};
+
   gtetrinet = callPackage ../games/gtetrinet {
     inherit (gnome2) GConf libgnome libgnomeui;
   };

